### ISSUE TITLE
Auto-build on push to Dev branch

### ISSUE THREAD

**Issue by LizzyTrickster** [LordFokas/MassiveOctoMustache#45]
Little fun project for me :D, tasks include:
- [ ] 1. Have a script that listens for commits/PR's pushed to Development branch (Either via GitHub POSTs or GitHub's notification API) by either @LordFokas or Myself containing the phrase *bot build this*
- [ ] 2. `git pull` the repo's dev branch into a local apache instance
- [ ] 3. `HTTP GET` the rebuild.php on the local webserver (or the PHP CLI)
- [ ] 4. `git commit` the changes and push up to GitHub

Normally this wouldn't be needed as people submitting changes should do the rebuild on their end before submitting a PR but for a simple change like a typo it can be useful.

**Comment by LordFokas** [comment]
the problem would be setting the changelog for the build...

**Comment by LizzyTrickster** [comment]
What do you mean?

**Comment by LordFokas** [comment]
I'm guessing you want the build system to update the website, right?

**Comment by LizzyTrickster** [comment]
I still don't know what you mean. All it will be doing is the same as if
you went to rebuild.php with it on a local Web server

**Comment by LordFokas** [comment]
yes, but running rebuild.php automatically doesn't change anything unless you change the data in /__data/* first, which means it'd be the same as running the script manually.

**Comment by LizzyTrickster** [comment]
Yes, that is entirely the point. All it allows for is so that if you make a quick typo correction / small change on the web, you can have the HTML regenerated without having to do it later

**Comment by LordFokas** [comment]
Oh I see what you mean now. Yeah makes sense.

**Comment by LizzyTrickster** [comment]
wow, this be old. I'm going to look into implementing this soon.

**Comment by LizzyTrickster** [comment]
This looks like it'd be fairly easy to impliment, a webhook needs to be set up with the push even to a url (which i'll discuss with @LordFokas elsewhere) and then from there I can get the commit author info (there is also committer info but if the commit was done from the web then that just gets set to the github stuff) and the commit message (which includes the extra comment as a newline) as well as the branch ref it was committed to.

It'll be locked down so that only myself and LF can trigger the building and even then it shouldn't need to be done all the time but it might also be useful if people forget to generate the html before they submit a PR
